### PR TITLE
Add copy path and copy relative path to context click menu

### PR DESCRIPTION
**Pull request by williamdes**
Release Notes:

- Added "Copy path" and "Copy relative path" to the file right click context menu

Related to https://github.com/zed-industries/zed/issues/4474

This replicates the tab behavior that exists on vscode
image

Tasks:
- [x] Check that it works on context menu in file
image
~~- [ ] Check that it works on a tab (does nothing)~~

### CODE REVIEW

File: crates/workspace/src/pane.rs [hunk 1]
```diff
@@ -1650,7 +1652,10 @@ impl Pane {
                                         );
                                     }),
                                 )
-                            });
+                            })
+                            .separator()
+                            .action("Copy Path", Box::new(CopyPath))
+                            .action("Copy Relative Path", Box::new(CopyRelativePath))
```
Comment (williamdes) [inline crates/workspace/src/pane.rs:1658]: I think I need help, I am not sure what is the way to implement this action in the workspace crate
It looks like each crate must implement actions

**Comment by williamdes** [comment]
I only implemented for the context menu in a file, I am unable to make it work for a tab. It is better to merge this one without the tab click. Maybe someone will make it or help me do it some day :)

**Comment by mikayla-maki** [comment]
It doesn't seem like VSCode includes this in it's mouse context menu:

Screenshot 2024-05-06 at 2 05 25 PM

As such, I don't think this is the right place for this action contextually. It should be on the tabs and the project panel.

**Comment by williamdes** [comment]
> As such, I don't think this is the right place for this action contextually. It should be on the tabs and the project panel.

I tend to agree but failed implementing it. 
Could someone let me know why my action was doing nothing?

**Comment by mikayla-maki** [comment]
Without a code sample or otherwise looking at the problem, I couldn't tell you. I suggest jumping in our discord and asking once you have one though :)

**Comment by williamdes** [comment]
Could you send me an invite link?

Here was my implementation https://github.com/zed-industries/zed/commit/91dc9a39bf7a06f2411567305d9a2f179b81c9f3

**Comment by mikayla-maki** [comment]
Hmmm, it should be available somewhere? Regardless, here's the link: https://discord.gg/29HzRq8V

**Comment by mattfbacon** [comment]
Sorry to see that this seems to be abandoned. Has there been newer work implementing this on tabs? I am missing this now. Should I open a new feature request issue?

**Comment by williamdes** [comment]
> Sorry to see that this seems to be abandoned. Has there been newer work implementing this on tabs? I am missing this now. Should I open a new feature request issue?

Yes please, if you can get this implemented it would be lovely. I really miss my patched version.

**Comment by mattfbacon** [comment]
See linked PR, I have the basic functionality but it doesn't work for all types of tabs yet.

**Pull request closed without merging**